projects/packages/paypal-payments/src/paypal-payment-buttons/class-paypal-payment-buttons.php

```diff
@@ -0,0 +1,177 @@
+		$code_body   = $attributes['codeBody'] ?? '';
+
+		if ( empty( $button_type ) || empty( $code_body ) ) {
+			return;
+		}
+
+		if ( 'stacked' === $button_type && ! empty( $code_head ) ) {
+			if ( preg_match( '/src="(https:\/\/(www\.)?(sandbox\.)?paypal\.com\/sdk\/js\?client-id=.+)"/', $code_head, $matches ) ) {
+				$script_url = esc_url( $matches[1] );
+				if ( ! empty( $script_url ) ) {
+					// We can't include the version number here. If we do, it is appended to the URL and causes a 400 response.
+					wp_enqueue_script( 'paypal-payment-buttons-block-head', $script_url, array(), null, false ); // phpcs:ignore WordPress.WP.EnqueuedResourceParameters.MissingVersion
+					add_filter(
+						'script_loader_tag',
+						function ( $tag, $handle, $src ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+							if ( 'paypal-payment-buttons-block-head' === $handle ) {
+								if ( false === strpos( $tag, 'data-paypal-partner-attribution-id' ) ) {
+									$tag = preg_replace( '/(\s+)src=([\'"])/', '$1 data-paypal-partner-attribution-id="WooNCPS_Ecom_Wordpress" src=$2', $tag );
+								}
+							}
+							return $tag;
+						},
+						10,
+						3
+					);
+				}
+			}
+		}
+
+		if ( 'single' === $button_type ) {
+			self::register_hooks();
+			$code_body = str_replace( 'inline-grid', 'grid', $code_body );
+			$attr_code = 'at_code=WooNCPS_Ecom_Wordpress';
+			if ( preg_match( '/\s+action=[\'"]([^\'"]+)[\'"]/', $code_body, $matches ) ) {
+				$action = esc_url( $matches[1] );
+				if ( ! empty( $action ) && false === strpos( $action, $attr_code ) ) {
+					$action    = add_query_arg( 'at_code', 'WooNCPS_Ecom_Wordpress', $action );
+					$code_body = preg_replace( '/(\s+action=[\'"])[^\'"]+([\'"])/', '$1' . $action . '$2', $code_body );
+				}
+			}
+		}
+
+		$allow_html = 'single' === $button_type ? array(
+			'style'   => array(),
+			'form'    => array(
+				'action' => array(),
+				'method' => array(),
+				'style'  => array(),
+			),
+			'input'   => array(
+				'class' => array(),
+				'type'  => array(),
+				'value' => array(),
+			),
+			'img'     => array(
+				'src'   => array(),
+				'alt'   => array(),
+				'style' => array(),
+			),
+			'section' => array(
+				'style' => array(),
+			),
+		) : array(
+			'div'    => array(
+				'id' => array(),
+			),
+			'script' => array(),
+		);
+
+		$code_body = wp_kses( $code_body, $allow_html );
+		return $code_body;
+	}
+
+	/**
+	 * Load editor styles for the block.
+	 * These are loaded via enqueue_block_assets to ensure proper loading in the editor iframe context.
+	 */
+	public static function load_editor_styles() {
+		$handle = 'jp-paypal-payments-ncps-blocks';
+
+		Assets::register_script(
+			$handle,
+			'../../dist/paypal-payment-buttons/editor.js',
+			__FILE__,
+			array(
+				'css_path'   => '../../dist/paypal-payment-buttons/editor.css',
+				'textdomain' => 'jetpack-paypal-payments',
+			)
+		);
+		wp_enqueue_style( $handle );
```

Comment:
I believe that if you set `'enqueue'    => true,` above, you wouldn't need to enqueue.

Edit: I see now that we enqueue the scripts separately. This may be worth clarifying in the docblock to avoid confusion?